### PR TITLE
Document debugging with Solr Admin Query Screen

### PR DESCRIPTION
**Pull request by yvanzo**
It summarizes tips provided by @alastair during a debug session.

The table of contents can be regenerated with:

    markdown-toc -i HACKING.md

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+From here, different combinations can be choosed depending on your needs:
+
+* Check `debugQuery` to have a detailed score computation (using search fields and boosts) for each result.
+
+* Set `fl` (field list) to
+  either `mbid,score` for the least detailed list of results
+  or `*,score` for the most detailed list of results
+
+* Restrict to a single result by setting “Raw Query Parameters” to:
```
Comment (alastair) [inline HACKING.md:103]: replace with:

If you expect another item to show up in the search results but it doesn't, you can ask solr to return the score logging for that item and compare it with the scores of the search results.

File: HACKING.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  `explainOther=mbid:fe78827f-e402-42a0-ab63-a3fd11c12cc4`
+
+Alternatively to get the same output format as in MB Search API
+(only since it is incompatible with showing any details),
+Unselect any value in the field `wt` (writer type) and
+set “Raw Query parameters” to either `wt=mbjson` or `wt=mbxml`.
+
```
Comment (alastair) [inline HACKING.md:110]: new section:

Some parameters are automatically set on all queries based on the core configuration. 

Some parameters are set based on the request handler: <https://github.com/metabrainz/mbsssss/blob/master/common/requestHandler-advanced.xml>. These are identified as "invariants", which means that the parameters will always be set to this value even if you specify a different value in the admin.

Some parameters for ranking are set with defaults and can be overridden: <https://github.com/metabrainz/mbsssss/blob/master/recording/conf/request-params.xml>
These can be changed by copying them into the relevant fields in the admin query panel to see the resulting change in ranking order.

File: HACKING.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  * [Query screen](#query-screen)
+
+<!-- tocstop -->
+
+Setup
+-----
+
+Set a development setup with [`musicbrainz-docker`](https://github.com/metabrainz/musicbrainz-docker).
+
+It allows to run any released or development version of
```
Comment (alastair) [inline HACKING.md:23]:
```suggestion
This can be used to run any released or development version of
```

File: HACKING.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
+The Debugger's Guide to MusicBrainz Solr
+========================================
+
+This file gives tips to debug MusicBrainz (advanced or not) indexed search based on Solr.
```
Comment (alastair) [inline HACKING.md:4]:
```suggestion
This file gives tips to debug MusicBrainz indexed search (basic and advanced) using Solr.
```

File: HACKING.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  either `mbid,score` for the least detailed list of results
+  or `*,score` for the most detailed list of results
+
+* Restrict to a single result by setting “Raw Query Parameters” to:
+  `explainOther=mbid:fe78827f-e402-42a0-ab63-a3fd11c12cc4`
+
+Alternatively to get the same output format as in MB Search API
+(only since it is incompatible with showing any details),
+Unselect any value in the field `wt` (writer type) and
+set “Raw Query parameters” to either `wt=mbjson` or `wt=mbxml`.
```
Comment (alastair) [inline HACKING.md:109]:
```suggestion
Alternatively to get the same output format as in MB Search API,m
Unselect any value in the field `wt` (writer type) and
set “Raw Query parameters” to either `wt=mbjson` or `wt=mbxml`.
These formats are incompatible with the `fl`, `debugQuery`, and `explainOther` flags, and setting
these options will not change the output.
```

**Review by alastair — COMMENTED** [review]
a few small changes, and a suggestion for a new section

File: HACKING.md (hunk 3 above)
Comment (yvanzo) [reply HACKING.md]: Applied as c124a0c

File: HACKING.md (hunk 5 above)
Comment (yvanzo) [reply HACKING.md]: Applied as b630fcc and amended with f966475

File: HACKING.md (hunk 2 above)
Comment (yvanzo) [reply HACKING.md:111]: Added as a new section “Parameters priority” with reformatted examples and a note about “Raw Query Parameters” having lesser priority than other form fields in commit 6b06a4c.

File: HACKING.md (hunk 1 above)
Comment (yvanzo) [reply HACKING.md]: Applied in commit 458125c

**Review by alastair — APPROVED** [review]
looks good, thanks!

**Pull request merged**